dandi/metadata.py

```diff
@@ -277,11 +277,12 @@ def extract_session(metadata: dict) -> list:
     for val in probe_ids:
         probes.append(models.Equipment(identifier=f"probe:{val}", name="Ecephys Probe"))
     probes = probes or None
-
+    if (metadata.get("session_id") or probes) is None:
+        return None
```

Comment:
I have missed the addition of this intimate relationship between "session" and "probe", in that a session without id would establish a session if there is any probe... doesn't smell right since `probe` is nohow special IMHO and any data, ecephys or icephys or neuroimaging acquisition does imply an "Acquisition session", but then why should we populate it (possibly with an empty id) if there are probes?  should we may be just make it 

```suggestion
    if metadata.get("session_id") is None:
        return None
```
here?

Reply:
probes are equipment used by the acquisition session (i changed the wording to acquisition session). since this is dealing with nwb, at this point this is the only id we have. as nwb improves this will all likely change.